Sources/PostgresNIO/New/Connection State Machine/ListenStateMachine.swift

```diff
@@ -224,12 +224,12 @@ extension ListenStateMachine {
         mutating func fail(_ error: Error) -> FailAction {
             switch self.state {
             case .initialized:
-                fatalError("Invalid state: \(self.state)")
+                return .none
```

Comment:
See https://github.com/vapor/postgres-nio/pull/487/files#r1647790890.
This is needed otherwise things will crash.

Reply:
can we get this fix into another pr with a dedicated unit test case and explanation why this can happen?

Reply:
hmmm sure. I'll remove the related test/changes from this PR.